File: GFG/Minimum characters to be added at front to make string palindrome/Minimum characters to be added at front to make string palindrome.java
class Solution {
    public static int minChar(String str) {
       //Write your code here
      int i =0, j = str.length()-1, last = j, ans =0;
      while(i < j){
          if(str.charAt(i) == str.charAt(j)){
              i++;
              j--;
          }else{
              ans++;
              i = 0;
              j = --last;
          }
      }
      return ans;
    }
    
}

//=====================================================
// class Solution {
//     public static int minChar(String s) {
//        //Write your code here
//        int i=0;
//        int j=s.length()-1;
//        int ans=0;
//        while(i<j){
//            if(s.charAt(i)==s.charAt(j)){
//                i++;
//                j--;
//            }else{
//                ans=s.length()-j;
//                i=0;
//                //reduce add
//                 while(s.charAt(i)==s.charAt(j)){
//                     ans--;
//                     i++;
//                 }
//                j--;
//            }
//        }
//        return ans;
//     }
// }

File: GFG/Minimum characters to be added at front to make string palindrome/Minimum characters to be added at front to make string palindrome.cpp
class Solution {
public:
    int minChar(string str){
        //Write your code here
        
        int i=0, j=str.length() - 1, last = j, ans = 0;
        
        while(i<j){
            
             //check for palindrome
             
             if(str[i] == str[j]){
                 
                 i++;
                 j--;
                 
             }
             
             //if characters dos not match decrease the value of last 
             //and put it into j , and increase value of ans'
             //start from 0
             
             else{
                 ans++;
                 i=0;
                 j= --last;
             }
            
            
        }
        
        //return ans
        
        return ans;
    }
};
